src/vehicle_move.cpp

```diff
@@ -469,7 +469,7 @@ void vehicle::thrust( int thd, int z )
         load = ( thrusting ? 1000 : 0 );
     }
     // rotorcraft need to spend 15% of load to hover, 30% to change z
-    if( is_rotorcraft() && is_flying_in_air() ) {
+    if( is_rotorcraft() && ( z > 0 || is_flying_in_air() ) ) {
```

Comment:
As far as I understand, the z level can be negative. What happens if you detonate a nuclear bomb, sit in a helicopter in an explosion crater and try to take off?

Reply:
Stall on liftoff due to radiometric effects, working as intended.